diff --git a/dist/index.js b/dist/index.js
index 913e0fad1f52e8a96b9ec0f635b3c9b7a11d9db4..082dc9b74bd71fa4f61c67c8183f91cd6738f982 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -6,7 +6,16 @@ async function compileSvg(source, compilerOptions) {
   if (compilerOptions.allow_props_children) {
     svgWithProps = svgWithProps.replace(/\{'\{'\}\s*(props\.children)\s*\{'\}'\}/g, "{$1}");
   }
-  return `export default (props = {}) => ${svgWithProps}`;
+  return `import { Portal } from "solid-js/web";
+export default (props = {}) => {
+  let div;
+  return (
+    <div ref={div}>
+      <Portal mount={div} useShadow={true}>${svgWithProps}</Portal>
+    </div>
+  );
+}
+`;
 }
 async function optimizeSvg(content, path, svgoConfig) {
   const config = svgoConfig || await loadConfig();
